### ISSUE TITLE
Prometheus software prerequisites

### ISSUE THREAD

**Issue by nealmcloughlin** [PSaaS-Developers/Project_issues#120]
The minimum and recommend software prequisites on firegrowthmodel.ca need to be reviewed and updated to reflect version 2021.12.03. Please review the requirements tab on https://firegrowthmodel.ca/pages/prometheus_software_e.html. Provide Franco with updates to make on the website.

**Comment by RobBryce** [comment]
It took less time to build this list, than to estimate how long it'd take to build this list.

PSaaS Ubuntu
-	Ubuntu 20.04
-	OpenJDK 11 JRE or later
-	deb installer will install for you
o	libxerces-c 3.2.2
o	libgdal 3.0.4
o	proj-bin 6.3.1-1
o	libc6 2.31
o	libstdc++6 10-20200411
o	libnuma1 2.0.12-1

PSaaS/Prometheus Windows
-	Windows 10 or higher
-	vcredist 2019 x64 (or latest)
-	vcredist 2012 x64
-	vcredist 2010 x64
-	Intel oneAPI DPC++/C++ Compiler Runtime 2022.0
-	Oracle Java 8 or later/OpenJDK 11 or later
-	.NET v4 (recommend latest)

**Comment by spydmobile** [comment]
The objective of this exercise is to provide the text block that needs to go on the website for Prometheus. This is ONLY about Prometheus that is ON the website now. 
You will need to do this again for the next version that makes it on the website too.

So for the current downloadable Prometheus:
**Q: What Prerequisites (Min versions and or recommended version) must the user have on windows to run the website version of Prometheus?**

**Comment by spydmobile** [comment]
@RobBryce ^^^^ that was meant for you ;-)

**Comment by RobBryce** [comment]
We aren't (yet) using Visual Studio 2022, so I see no need to suggest Win11.  It works on Win11, though.  Win10 is marked for end of support on Oct 14, 2025, so that's 3 years off.  https://learn.microsoft.com/en-us/lifecycle/products/windows-10-home-and-pro

We are not listing prerequisites for Java 8, though we do check them in the installer.  What do you want to do?

The current build of Java 8 is 333 so that can be updated.  Java 8 was released about 8 years ago, but if I recall, we cannot update to a more current version because of agency requirements (at the time, may have changed).  It would be nice to update the version of Java to something more current.

The link to the Intel library needs updated.  On firegrowthmodel.ca, the download of 2021.3.0.3372 is available since Intel doesn't have it on their site.  Each time I build a compilation environment, though, the version of OneAPI that I download has been updated, and I can't find a consistent version history of it.  There are changes happening, noted by compiler options disappearing, etc.  On this note, we haven't kept the installer current for versions of Intel's compiler either, since it changes to GUID with each formal release (which is why you provide the download for that prerequisite).

I'm comfortable with the hardware requirements.  We could bump the cores to 6 hyperthreading cores, but we've found that performance degrades past the current recommendation (but Prometheus will now limit itself to the # of cores anyway, to prevent that degradation).  If a power user like Brett runs multiple instances of Prometheus at once, then more will be needed.  I'd also suggest going to 8GB and 16GB RAM.

So, most of the these changes are particularly necessary but may be appropriate (such as the update to Intel to match the site's download link), and I'm not sure what you want to do about Java prerequisites not being listed.

**Comment by spydmobile** [comment]
@RobBryce we are looking for you to update the requirements list  found on the fgm.ca website from the requirements tab found here:
https://firegrowthmodel.ca/pages/prometheus_software_e.html

Those requirements provide a minimum and a recommended, both need updating.

Reading above I get the sense you are still looking for more direction.

`We are not listing prerequisites for Java 8, ` but yes It looks like we are doing exactly that, so I dont understand your point here. I agree its bad to only list a locked version, we are not implementing agency requirements here.... further, we are supposed to support OpenJDK for this as well so additonally what versions of that for min and rec?

`The link to the Intel library needs updated` updated to what updated where?

`I'm comfortable with the hardware requirements. We could bump the cores to 6 hyperthreading cores` so min 4 hyper-threading cores and reccomended 6 hyper-threading cores?

`I'd also suggest going to 8GB and 16GB RAM.` So min 6 rec 16?

Last paragraph is not clear, please explain it again.

**Comment by RobBryce** [comment]
@spydmobile - the format upon which to provide the updates was not specified, so we provided it in a manner consistent with this in the past, and the most economical manner possible.  It now sounds like you want an updated table in HTML format but I'm not sure about that, and it's unclear what other aspects of that table definition are needed for basic insertion.

I also provided some commentary on items that I think need updating but require discussion and input due to out of date versions of Java, etc.  This is outside simply providing items for table entries, since an update to a newer version of Java (for example) requires input from key agencies, and the table isn't reflective of modern commonly available hardware that we could expect users to use.

I think at this point, the opportunity to minimize costs with simply providing updates that can be copy/pasted into the existing table has passed, because that's what I've tried to do.  I'll provide a quotation and estimate once I know specifically what is being asked for here.

**Comment by spydmobile** [comment]
@RobBryce No thanks. Dont think that is what we need. You just need to provide the updated info to replace the outdated info on the website, other concerns should be addressed in separate issues unless they directly affect the existing version on the website.

**Comment by nealmcloughlin** [comment]
@RobBryce I'm asking that you verify that system requirements on the website are aligned with version 2021.12.03 which is the most current version available for download. We did not update the system requirements when we updated the website with version 2021.12.03. Perhaps nothing has changed but I want to be sure what we are documenting the correct system requirements for version 2021.12.03. This is not a discussion regarding system requirements going forward. Franco will update the website so you can specify any changes here in Zenhub. 
Thanks.

**Comment by BadgerOnABike** [comment]
You are right, we did not upgrade the requirements. To the point that they are currently incorrect.

**Comment by spydmobile** [comment]
To be clear, we need Minimum Required Specs and Prerequisites and Recommended Specs and Prerequisites

Ok so would this be correct then?

# Prometheus 2021.12.03 

Requirement | Minimum | Recommended
-- | -- | --
Operating System* | 64-bit Windows 10 or Windows Server 2008 R2 | Windows 11
Microsoft Software | .NET Framework 4.7 and Visual C++ 2017 Redistributable (x64) | .NET Framework 4.x (Latest) and Visual C++ 2017 Redistributable (x64)
Oracle Software | Java 8.0 for Windows (x64)(Update 151) | Java 8.0 for Windows (x64)(Update 191)
Intel® Software | 2019 C++ Redistributable Library for Windows (x64)(Update 1: 19.1.144) | 2019 C++ Redistributable Library for Windows (x64)(Latest)
Processors | 4 hyper-threading cores | 6 hyper-threading cores
RAM | 8 GB | 16 GB
Hard Drive | 5 GB | 500 GB
Screen Resolution | 1280 x 720 | 1600 x 1024

**Comment by RobBryce** [comment]
This appears correct for the 2021 build mentioned.  But for Intel software, I would just make them the same, and possibly link to your copy of that redistributable.  The Intel link will need updated for the current build.  Do you want links for download pages, or are they sufficient to leave embedded in the installer?

**Comment by spydmobile** [comment]
@RobBryce 

> This appears correct for the 2021 build mentioned. 

Ok Thanks.

> But for Intel software, I would just make them the same, and possibly link to your copy of that redistributable. 

Ok I will link to https://spyd.com/fgm.ca/libs/w_dpcpp_cpp_runtime_p_2021.3.0.3372.exe

> The Intel link will need updated for the current build. 

What "Current Build" are you talking about? 

> Do you want links for download pages, or are they sufficient to leave embedded in the installer?

I think we can leave them in the installer until EOL at which point we can link to our own copies to keep them indefinitely, do you agree?

**Comment by RobBryce** [comment]
For "current build", I'm referring to Intel's ever-changing target and website, and I believe this is the current correct link: https://www.intel.com/content/www/us/en/developer/articles/tool/compilers-redistributable-libraries-by-version.html

The very first link for OneAPI DPC++/C++ will be needed.

Note that we currently aren't using this version for the latest builds because of how Intel keeps changing GUIDs.  We have the (discussed) forwards-compatibility issue.

For the other Q: it may be best to link to the project's download pages since Intel keeps changing stuff - we have control over future-proofing.  Then, that link can be updated as needed, and we control that.

**Comment by RobBryce** [comment]
We have done work on this item, but due to the nature and history of this work item, I'm expecting that there will be further revisions.  Please review and we'll act accordingly.

**Comment by spydmobile** [comment]
@RobBryce we are going to need this updated for the release, please let me know when we can update the 1.0 requirements (Logically after there is a 1.0)

**Comment by spydmobile** [comment]
@rob meaning we will need this updated for WISE (and later for Prometheus EOL)
